### ISSUE TITLE
Cargo pulling in dependencies cfg-ed for other platforms in nightly

### ISSUE THREAD

**Issue by GregoryConrad** [rust-lang/rust#121911]
I started noticing this happening in my CI on [February 24th](https://github.com/GregoryConrad/rearch-rs/actions/runs/8029840032/attempts/1); everything is ok on stable, but the following occurs in nightly:
https://github.com/GregoryConrad/rearch-rs/actions/runs/8123967250/job/22204944957?pr=37

Cargo is complaining of:

```
$ cargo +nightly clippy
...
error: multiple versions for dependency `windows-sys`: 0.48.0, 0.52.0
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions
  = note: `-D clippy::multiple-crate-versions` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::multiple_crate_versions)]`

error: multiple versions for dependency `windows-targets`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_aarch64_gnullvm`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_aarch64_msvc`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_i686_gnu`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_i686_msvc`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_x86_64_gnu`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_x86_64_gnullvm`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: multiple versions for dependency `windows_x86_64_msvc`: 0.48.5, 0.52.4
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

error: could not compile `rearch-tokio` (lib) due to 9 previous errors
```

But that makes no sense, since those dependencies are behind a `cfg(windows)` over in tokio. In fact, running
`cargo +nightly tree --target=x86_64-unknown-linux-gnu` yields no windows related dependencies as expected.

Thus, it looks likely cargo is pulling in dependencies behind a `cfg(windows)` flag on other platforms, at least for clippy.

**Comment by GregoryConrad** [comment]
To reproduce, simply:

```bash
git clone https://github.com/GregoryConrad/rearch-rs
cd rearch-rs
cargo clippy
cargo +nightly clippy
```

On Ubuntu or macOS.

**Comment by Nilstrieb** [comment]
this issue should be moved to the rust-lang/cargo repository
